doc/api/v8.md

```diff
@@ -1051,6 +1051,51 @@ added:
 
 Returns true if the Node.js instance is run to build a snapshot.
 
+### Class: `v8.MemoryProfileWatchdog`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+#### `new MemoryProfileWatchdog([options])`
+
+* `interval` {number} How often the watchdog polls.
+* `maxRss` {number} Node.js will take snapshot when rss is
+  greater than or equal to `maxRss`.
+* `maxUsedHeapSize` {number} Node.js will take snapshot when used heap size is
+  greater than or equal to `maxUsedHeapSize`.
+* `filename` {string} Node.js will write heap snapshot to `filename`.
+
+Creates a new `MemoryProfileWatchdog` object, `maxRss` or `maxUsedHeapSize` must
+be setted. `MemoryProfileWatchdog` will create a thread and poll the memory
+info periodically, when the rss or used heap size is greater than or equal to
+the value you set, Node.js will write a heap snapshot to `filename` then stop
```

Comment:
I think writing a heap snapshot is a blocking operation therefore this may have significant impact to user application responsivness.
As far as I know v8 requires quite some extra memory to write a heap snapshot. Therefore better mention that this shoudl be configured with enough memory left on the machine otherwise the watchdog might trigger an OOM killer or similar.

Reply:
Thanks, i will add it to docs.

Reply:
Done!